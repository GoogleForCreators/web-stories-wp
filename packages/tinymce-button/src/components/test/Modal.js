/*
 * Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/**
 * External dependencies
 */
import renderer from 'react-test-renderer';
/**
 * Internal dependencies
 */
import WebStoriesModal from '../Modal';

jest.mock('@wordpress/data', () => ({
  select: jest.fn(() => ({
    getCurrentView: () => 'grid',
  })),
}));

jest.mock('@wordpress/i18n', () => ({
  // eslint-disable-next-line no-unused-vars
  __: (val, domain) => val,
}));

jest.mock('@wordpress/components', () => ({
  Modal: 'Modal',
  RangeControl: 'Range',
  SelectControl: 'Select',
  Button: 'Button',
  TextControl: 'TextControl',
}));

jest.mock('../../utils/globals', () => ({
  webStoriesData: {
    views: [],
  },
}));

jest.mock('../../components/controls/Toggle', () => 'TinyMCEToggle');

jest.mock('../../utils', () => ({
  updateViewSettings: jest.fn((settings) => settings),
}));

describe('TinyMCE Controls Modal', () => {
  it('modal is opened and view is non-circle', () => {
    const props = {
      modalOpen: true,
      settings: {
        archive_link: {
          show: true,
        },
      },
      prepareShortCode: jest.fn(),
    };

    const tree = renderer.create(<WebStoriesModal {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('modal is opened and view is circle', () => {
    const props = {
      modalOpen: true,
      settings: {
        archive_link: {
          show: true,
        },
      },
      prepareShortCode: jest.fn(),
    };

    const tree = renderer.create(<WebStoriesModal {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
