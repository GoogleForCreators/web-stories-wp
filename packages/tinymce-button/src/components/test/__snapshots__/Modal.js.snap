// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TinyMCE Controls Modal modal is opened and view is circle 1`] = `
<Modal
  className="component_web_stories_mce_model"
  closeButtonLabel="Close"
  onRequestClose={[Function]}
  shouldCloseOnClickOutside={false}
  title="Web Stories"
>
  <Select
    label="Select Layout"
    onChange={[Function]}
    options={Array []}
  />
  <Range
    label="Number of Stories"
    max={20}
    min={1}
    onChange={[Function]}
  />
  <Select
    label="Order By"
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "Date",
          "value": "post_date",
        },
        Object {
          "label": "Title",
          "value": "post_title",
        },
      ]
    }
  />
  <Select
    label="Order"
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "Ascending",
          "value": "ASC",
        },
        Object {
          "label": "Descending",
          "value": "DESC",
        },
      ]
    }
  />
  <TinyMCEToggle
    field="title"
  />
  <TinyMCEToggle
    field="excerpt"
  />
  <TinyMCEToggle
    field="author"
  />
  <TinyMCEToggle
    field="date"
  />
  <TinyMCEToggle
    field="sharp_corners"
  />
  <TinyMCEToggle
    field="archive_link"
    fieldObj={
      Object {
        "show": true,
      }
    }
  />
  <TextControl
    label="Archive Link Label"
    onChange={[Function]}
  />
  <div
    className="alignright"
    style={
      Object {
        "padding": "20px 0",
      }
    }
  >
    <Button
      isPrimary={true}
      onClick={[Function]}
    >
      Insert
    </Button>
    <Button
      onClick={[Function]}
    >
      Cancel
    </Button>
  </div>
</Modal>
`;

exports[`TinyMCE Controls Modal modal is opened and view is non-circle 1`] = `
<Modal
  className="component_web_stories_mce_model"
  closeButtonLabel="Close"
  onRequestClose={[Function]}
  shouldCloseOnClickOutside={false}
  title="Web Stories"
>
  <Select
    label="Select Layout"
    onChange={[Function]}
    options={Array []}
  />
  <Range
    label="Number of Stories"
    max={20}
    min={1}
    onChange={[Function]}
  />
  <Select
    label="Order By"
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "Date",
          "value": "post_date",
        },
        Object {
          "label": "Title",
          "value": "post_title",
        },
      ]
    }
  />
  <Select
    label="Order"
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "Ascending",
          "value": "ASC",
        },
        Object {
          "label": "Descending",
          "value": "DESC",
        },
      ]
    }
  />
  <TinyMCEToggle
    field="title"
  />
  <TinyMCEToggle
    field="excerpt"
  />
  <TinyMCEToggle
    field="author"
  />
  <TinyMCEToggle
    field="date"
  />
  <TinyMCEToggle
    field="sharp_corners"
  />
  <TinyMCEToggle
    field="archive_link"
    fieldObj={
      Object {
        "show": true,
      }
    }
  />
  <TextControl
    label="Archive Link Label"
    onChange={[Function]}
  />
  <div
    className="alignright"
    style={
      Object {
        "padding": "20px 0",
      }
    }
  >
    <Button
      isPrimary={true}
      onClick={[Function]}
    >
      Insert
    </Button>
    <Button
      onClick={[Function]}
    >
      Cancel
    </Button>
  </div>
</Modal>
`;
